sdks/java/core/src/main/java/org/apache/beam/sdk/values/BeamRecord.java

```diff
@@ -282,7 +283,7 @@ public int getFieldCount() {
    * Return the list of data values.
    */
   public List<Object> getDataValues() {
-    return dataValues;
+    return ImmutableList.copyOf(dataValues);
```

Comment:
I'm afraid this line will throw errors, as `ImmutableList` doesn't allow NULL values.